### DISCUSSION TITLE
suggestion: support object syntax for Form::labelPosition (responsive layout)

### DISCUSSION THREAD

**Discussion by GammaWolf** [adobe/react-spectrum#2414] [Ideas]
It may be useful if one could use the responsive layout mechanics described [here](https://react-spectrum.adobe.com/react-spectrum/layout.html#responsive-layout) to be able to switch between top and side `Form` labeling (property `labelPosition`) depending on a breakpoint. The top labeling is less wide and thus better suited for mobile screens. The side labeling looks better (to me) if it can fit. Currently the property supports runtime changing (tested it with a button toggling a useState var) but not the responsive object syntax.

**Comment by snowystinger** [comment]
Is this what you need?
https://codesandbox.io/s/snowy-leaf-fq60f?file=/src/App.js
I'm just importing useBreakpoints which uses the context from the BreakpointProvider, then use that to determine what prop value to set.

**Reply by GammaWolf** [reply]
That works well, thank you.
